reconcile/change_owners/change_log_tracking.py

```diff
@@ -0,0 +1,69 @@
+import logging
+
+from reconcile.change_owners.bundle import (
+    NoOpFileDiffResolver,
+    QontractServerDiff,
+)
+from reconcile.change_owners.change_owners import fetch_change_type_processors
+from reconcile.change_owners.change_types import ChangeTypeContext
+from reconcile.change_owners.changes import aggregate_file_moves, parse_bundle_changes
+from reconcile.utils import gql
+from reconcile.utils.runtime.integration import NoParams, QontractReconcileIntegration
+from reconcile.utils.semver_helper import make_semver
+from reconcile.utils.state import init_state
+
+
+class ChangeLogIntegration(QontractReconcileIntegration[NoParams]):
+    def __init__(self) -> None:
+        super().__init__(NoParams())
+        self.qontract_integration = "change-log-tracking"
+        self.qontract_integration_version = make_semver(0, 1, 0)
+
+    @property
+    def name(self) -> str:
+        return self.qontract_integration
+
+    def run(self, dry_run: bool) -> None:
+        change_type_processors = [
+            ctp
+            for ctp in fetch_change_type_processors(
+                gql.get_api(), NoOpFileDiffResolver()
+            )
+            if ctp.labels and "change_log_tracking" in ctp.labels
+        ]
+        state = init_state(
+            integration=self.name,
+        )
+        int_state_path = state.state_path
+        state.state_path = "bundle-archive/diff"
+        change_log: dict[str, list[dict]] = {}
+        for item in state.ls():
+            key = item.lstrip("/")
+            commit = key.rstrip(".json")
+            logging.info(f"Processing commit {commit}")
+            change_log.setdefault(commit, [])
+            obj = state.get(key, None)
```

Comment:
This means we always need to process all diff objects in s3 one by one, can we load saved `bundle-diffs.json` first then only process missing commits?
Then `bundle-diffs.json` can keep all diffs even diff objects deleted after expire, we can also include some metadata field like timestamp to help reduce size if needed later, or delete commits when diff no longer exists.

Reply:
this is a great idea. added a click.option implementation in https://github.com/app-sre/qontract-reconcile/pull/4620/commits/3f6215f963f1477834130abed7a0c94fafda0704 in case we need to re-process at some point.

Reply:
nice! a few performance and edge case handling:

* first time `integration.get` may fail if the file not exist
* ChangeLog.items can be a dict, key is commit, value is ChangeLogItem for faster lookup, as the number of commits can be huge
* `if not self.params.process_existing:` I think you mean `if self.params.process_existing:`
* can skip upload to s3 if `change_log` is same as before